custom/enikshay/management/commands/investigations_reconciliation.py

```diff
@@ -0,0 +1,258 @@
+DATE_MODIFIED_FIELD = "modified_on"
+PROPERTIES_TO_BE_COALESCED = [
+    "lft_results",
+    "blood_urea_results",
+    "other_results",
+    "tsh_results",
+    "s_cr_results",
+    "serum_lipase_results",
+    "audiogram_results",
+    "urinegravindex_results",
+    "ecgqtc_results",
+    "electrolyte_results",
+    "upt_results",
+    "cbcplatelets_results",
+    "culture_date",
+    "culture_lab_serial_number",
+    "culture_result_value",
+]
+
+
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument('--dry_run', action='store_true')
+        parser.add_argument('--recipient', type=str)
+
+    def handle(self, *args, **options):
+        # self.dry_run = options.get('dry_run')
+        self.dry_run = True
+        self.recipient = (options.get('recipient') or 'mkangia@dimagi.com')
+        self.recipient = list(self.recipient) if not isinstance(self.recipient, basestring) else [self.recipient]
+        self.result_file_name = self.setup_result_file()
+        self.case_accessor = CaseAccessors(DOMAIN)
+        self.investigation_interval_values = []
+        with open(self.result_file_name, 'a') as csvfile:
+            self.writer = csv.DictWriter(csvfile, fieldnames=self.get_result_file_headers())
+            # iterate all person cases
+            for person_case_id in self._get_open_person_case_ids_to_process():
+                person_case = self.case_accessor.get_case(person_case_id)
+                # check if person is a public app case
+                if self.public_app_case(person_case):
+                    # get all confirmed drtb cases to check for cases under them
+                    open_confirmed_drtb_episode_cases = get_open_confirmed_drtb_episode_cases(person_case_id)
+                    for episode_case in open_confirmed_drtb_episode_cases:
+                        episode_case_id = episode_case.case_id
+                        # get any investigations that need to be reconciled under this episode
+                        investigations_to_be_reconciled = self.episode_case_needs_reconciliation(episode_case)
+                        if investigations_to_be_reconciled:
+                            for interval_type, investigation_cases in investigations_to_be_reconciled.items():
+                                self.reconcile_investigation_cases(episode_case_id, investigation_cases, csvfile)
+
+        self.email_report()
+
+    def email_report(self):
+        csvfile = open(self.result_file_name)
+        email = EmailMessage(
+            subject="Investigations Reconciliation Report",
+            body="Please find attached report for a %s run finished at %s." %
+                 ('dry' if self.dry_run else 'real', datetime.now()),
+            to=self.recipient,
+            from_email=settings.DEFAULT_FROM_EMAIL
+        )
+        email.attach(filename=self.result_file_name, content=csvfile.read())
+        csvfile.close()
+        email.send()
```

Comment:
There's a lot of repeated code in these migrations, it would be nice to pull some of that out if possible.

Reply:
Didn't want to extract the code and put in the app so,
adding BaseModelReconciliationCommand and then inherit that.